lib/puma/configuration.rb

```diff
@@ -338,6 +343,16 @@ def self.temp_path
 
     private
 
+    def require_processor_counter
+      require 'concurrent/utility/processor_counter'
+    rescue LoadError
+      warn <<~MESSAGE
+        WEB_CONCURRENCY=auto requires the "concurrent-ruby" gem to be installed.
+        Please add "concurrent-ruby" to your Gemfile.
```

Comment:
That works for me

Reply:
I also created a commit https://github.com/codergeek121/rails/commit/ccb9a8e2117740066b188925844652427b585b48 to update Rails default values & docs to include this new `auto` config value. I'll open up a PR against Rails, if the `auto` configuration is merged in puma 👍